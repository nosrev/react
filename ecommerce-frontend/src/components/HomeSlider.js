import React from "react";
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import slide1 from '../img/slide1.png';
import { ReactComponent as ArrowLeft } from '../svg/arrow-left.svg';
import { ReactComponent as ArrowRight } from '../svg/arrow-right.svg';

const HomeSlider = () => {
    const content = [
        {
            image: slide1,
            title: 'Fashion',
            subtitle: 'Lançamentos',
            button: 'Comprar',
            bg: 'black'
        },
        {
            image: slide1,
            title: 'Casual',
            subtitle: 'Promoções',
            button: 'Comprar',
            bg: 'primary'
        },
    ];

    function PrevArrow(props) {
      const { className, style, onClick } = props;
      return (
        <div
          className={`${className} go-previous`}
          style={style}
          onClick={onClick}
        >
            <ArrowLeft className="icon w-8 fill-white" />
        </div>
      );
    }

    function NextArrow(props) {
      const { className, style, onClick } = props;
      return (
        <div
          className={`${className} go-next`}
          style={style}
          onClick={onClick}
        >
            <ArrowRight className="icon w-8 fill-white" />
        </div>
      );
    }

    const settings = {
        dots: false,
        infinite: true,
        slidesToShow: 1,
        slidesToScroll: 1,
        speed: 200,
        fade: true,
        nextArrow: <NextArrow />,
        prevArrow: <PrevArrow />
    };

    return (
        <div id="slider-wrapper">
            <div className="slider">
                <Slider {...settings}>
                    {content.map((item, index) => (
                        <div className={"bg-" + item.bg } key={index}>
                            <img src={item.image} className="slide-img" alt={item.title} />
                            <p className="title text-blue2">{item.title}</p>
                            <p className="subtitle text-white">{item.subtitle}</p>
                            <a href="/" className="btn-view inline-block bg-primary text-white border-2 border-primary hover:bg-white hover:text-primary">{item.button}</a>
                        </div>
                    ))}
                </Slider>
            </div>
        </div>
    )
}

export default HomeSlider;
