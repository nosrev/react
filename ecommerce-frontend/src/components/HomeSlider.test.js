import renderer from 'react-test-renderer';
import HomeSlider from './HomeSlider';

it('render HomeSlider', () => {
  const component = renderer.create(
    <HomeSlider />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
