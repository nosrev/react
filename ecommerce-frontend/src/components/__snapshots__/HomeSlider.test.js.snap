// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render HomeSlider 1`] = `
<div
  id="slider-wrapper"
>
  <div
    className="slider"
  >
    <div
      className="slick-slider slick-initialized"
      dir="ltr"
    >
      <div
        className="slick-arrow slick-prev go-previous"
        onClick={[Function]}
        style={
          Object {
            "display": "block",
          }
        }
      >
        <svg
          className="icon w-8 fill-white"
        >
          arrow-left.svg
        </svg>
      </div>
      <div
        className="slick-list"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={null}
        onMouseMove={null}
        onMouseUp={[Function]}
        onTouchCancel={null}
        onTouchEnd={[Function]}
        onTouchMove={null}
        onTouchStart={[Function]}
        style={Object {}}
      >
        <div
          className="slick-track"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <div
            aria-hidden={false}
            className="slick-slide slick-active slick-current"
            data-index={0}
            onClick={[Function]}
            style={
              Object {
                "left": -0,
                "opacity": 1,
                "outline": "none",
                "position": "relative",
                "transition": "opacity 200ms ease, visibility 200ms ease",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="bg-black"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt="Fashion"
                  className="slide-img"
                  src="slide1.png"
                />
                <p
                  className="title text-blue2"
                >
                  Fashion
                </p>
                <p
                  className="subtitle text-white"
                >
                  Lançamentos
                </p>
                <a
                  className="btn-view inline-block bg-primary text-white border-2 border-primary hover:bg-white hover:text-primary"
                  href="/"
                >
                  Comprar
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={1}
            onClick={[Function]}
            style={
              Object {
                "left": -0,
                "opacity": 0,
                "outline": "none",
                "position": "relative",
                "transition": "opacity 200ms ease, visibility 200ms ease",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="bg-primary"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt="Casual"
                  className="slide-img"
                  src="slide1.png"
                />
                <p
                  className="title text-blue2"
                >
                  Casual
                </p>
                <p
                  className="subtitle text-white"
                >
                  Promoções
                </p>
                <a
                  className="btn-view inline-block bg-primary text-white border-2 border-primary hover:bg-white hover:text-primary"
                  href="/"
                >
                  Comprar
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="slick-arrow slick-next go-next"
        onClick={[Function]}
        style={
          Object {
            "display": "block",
          }
        }
      >
        <svg
          className="icon w-8 fill-white"
        >
          arrow-right.svg
        </svg>
      </div>
    </div>
  </div>
</div>
`;
